static/app/api.tsx

```diff
@@ -243,7 +243,10 @@ type ClientOptions = {
    * The base URL path to prepend to API request URIs.
    */
   baseUrl?: string;
-
+  /**
+   * Credentials policy to apply to each request
+   */
+  credentials?: RequestCredentials;
```

Comment:
can we split this out into a diff PR?

This has a bit of security implications here so we probably want to review this on it's own